CHANGELOG.md

```diff
@@ -0,0 +1,157 @@
+interactions, and protect against several kinds of common bugs that occurred in
+Objective-C.
+
+### Interrupted event
+
+In addition to the `Next`, `Error`, and `Completed` events that have always been
+part of RAC, version 3.0 [adds another terminating
+event](https://github.com/ReactiveCocoa/ReactiveCocoa/pull/1735)—called
+`Interrupted`—that is used to communicate cancellation.
+
+Now, whenever a [producer](#cold-signals-are-now-signalproducers) is disposed
+of, one final `Interrupted` event will be sent to all consumers, giving them
+a chance to react to the cancellation.
+
+Similarly, observing a [hot signal](#hot-signals-are-now-signals) that has
+already terminated will immediately result in an `Interrupted` event, to clearly
+indicate that no further events are possible.
+
+This brings disposal semantics more in line with normal event delivery, where
+events propagate downstream from producers to consumers. The result is a simpler
+model for reasoning about non-erroneous, yet unsuccessful, signal terminations.
+
+**Note:** Custom `Signal` and `SignalProducer` operators should handle any received
+`Interrupted` event by forwarding it to their own observers. This ensures that
+interruption correctly propagates through the whole signal chain.
+
+### Objective-C bridging
+
+To support interoperation between the Objective-C APIs introduced in RAC 2 and
+the Swift APIs introduced in RAC 3, the framework offers [bridging
+functions](ReactiveCocoa/Swift/ObjectiveCBridging.swift) that can convert types
+back and forth between the two.
+
+Because the APIs are based on fundamentally different designs, the conversion is
+not always one-to-one; however, every attempt has been made to faithfully
+translate the concepts between the two APIs (and languages).
+
+**Common conversions include:**
+
+* The `RACSignal.toSignalProducer` method **†**
+    * Converts `RACSignal *` to `SignalProducer<AnyObject?, NSError>`
+* The `toRACSignal()` function
+    * Converts `SignalProducer<AnyObject?, ErrorType>` to `RACSignal *`
+    * Converts `Signal<AnyObject?, ErrorType>` to `RACSignal *`
+* The `RACCommand.toAction` method **‡**
+    * Converts `RACCommand *` to `Action<AnyObject?, AnyObject?, NSError>`
+* The `toRACCommand` function **‡**
+    * Converts `Action<AnyObject?, AnyObject?, ErrorType>` to `RACCommand *`
+
+**†** It is not possible (in the general case) to convert arbitrary `RACSignal`
+instances to `Signal`s, because any `RACSignal` subscription could potentially
+involve side effects. To obtain a `Signal`, use `RACSignal.toSignalProducer`
+followed by `SignalProducer.start`, thereby making those side effects explicit.
+
+**‡** Unfortunately, the `executing` properties of actions and commands are not
+synchronized across the API bridge. To ensure consistency, only observe the
+`executing` property from the base object (the one passed _into_ the bridge, not
+retrieved from it), so updates occur no matter which object is used for
+execution.
```

Comment:
:scream: That seems terrible for composition. A bridged action or command can't be passed around to other objects because they don't fulfill their contracts.